spring-boot-project/spring-boot/src/main/java/org/springframework/boot/SpringApplication.java

```diff
@@ -802,22 +803,20 @@ private void callRunner(CommandLineRunner runner, ApplicationArguments args) {
 	private void handleRunFailure(ConfigurableApplicationContext context, Throwable exception,
 			SpringApplicationRunListeners listeners) {
 		try {
-			try {
-				handleExitCode(context, exception);
-				if (listeners != null) {
-					listeners.failed(context, exception);
-				}
-			}
-			finally {
-				reportFailure(getExceptionReporters(context), exception);
-				if (context != null) {
-					context.close();
-				}
+			handleExitCode(context, exception);
+			if (listeners != null) {
+				listeners.failed(context, exception);
 			}
 		}
 		catch (Exception ex) {
 			logger.warn("Unable to close ApplicationContext", ex);
 		}
+		finally {
+			reportFailure(getExceptionReporters(context), exception);
+			if (context != null) {
+				context.close();
```

Comment:
This changes behaviour as any exception from the `close()` call will no longer be caught and logged as a warning.